src/cuda_utils.cc

```diff
@@ -83,39 +83,41 @@ GetDeviceMemoryInfo(const int device_id, size_t* free, size_t* total)
 }
 
 Status
-EnablePeerAccess(const double min_compute_capability)
+EnablePeerAccess(const double min_compute_capability, bool enable_peer_access)
 {
 #ifdef TRITON_ENABLE_GPU
   // If we can't enable peer access for one device pair, the best we can
   // do is skipping it...
-  std::set<int> supported_gpus;
-  bool all_enabled = false;
-  if (GetSupportedGPUs(&supported_gpus, min_compute_capability).IsOk()) {
-    all_enabled = true;
-    int can_access_peer = false;
-    for (const auto& host : supported_gpus) {
-      auto cuerr = cudaSetDevice(host);
-
-      if (cuerr == cudaSuccess) {
-        for (const auto& peer : supported_gpus) {
-          if (host == peer) {
-            continue;
+  if (enable_peer_access) {
+    std::set<int> supported_gpus;
+    bool all_enabled = false;
+    if (GetSupportedGPUs(&supported_gpus, min_compute_capability).IsOk()) {
+      all_enabled = true;
+      int can_access_peer = false;
+      for (const auto& host : supported_gpus) {
```

Comment:
small naming tweak - is `device` better than `host` ? `host` implies the cpu  (i.e. host memory)

Reply:
In this context, it is host v.s. peer, can use different naming but the new naming should highlight the idea of "current" device and "peer" device. To enable peer access between two devices, the call needs to be done in both direction